packages/component/src/loadableModulesMap.js

```diff
@@ -0,0 +1,36 @@
+let federatedModuleMap = {};
+const localModuleNames = [];
+
+const isFederated = (loadableCtor) => loadableCtor.resolve().indexOf('webpack/container/remote') === 0;
+
+export const register = (loadableCtor) => {
+  const moduleName = loadableCtor.chunkName();
+  if (isFederated(loadableCtor)) {
+    federatedModuleMap[moduleName] = loadableCtor.importAsync;
+  } else {
+    localModuleNames.push(moduleName);
+  }
+};
+
+/**
+ * Calls importAsync for all the modules listed in moduleNames
+ * Resolves after modules are loaded
+ */
+export const preload = (moduleNames) => {
+  const federatedModules = moduleNames
+    .filter((name) => !localModuleNames.includes(name))
+    .map((name) => federatedModuleMap[name]);
+
+  // only need to preload module once
+  federatedModuleMap = {};
+
+  return Promise.all(federatedModules.map((loader) => loader()));
+};
+
+
+/**
+ * Preload all the existing federated modules, to be on the server (we don't have loadable required chunks on SSR)
+ */
+export const preloadAll = () => preload(Object.keys(federatedModuleMap))
```

Comment:
I like this idea. Pretty usable.

Reply:
Still not sure if it's better than static import for server side. Because static import creates less chunks and makes SSR easier